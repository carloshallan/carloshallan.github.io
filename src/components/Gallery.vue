<template>
  <div class="gallery">
    <router-link to="work" tag="div" class="row">
      <figure>Image</figure>
      <header>
        <h2>Title 1</h2>
        <p>test 1</p>
      </header>
    </router-link>
    <router-link to="work" tag="div" class="row">
      <figure>Image</figure>
      <header>
        <h2>Title 2</h2>
        <p>test 2</p>
      </header>
    </router-link>
  </div>
</template>
<script lang="ts">
import Vue, { PropType } from 'vue'
import { GalleryItems } from '@/types'

export default Vue.extend({
  name: 'Gallery',
  props: {
    galleryItems: Array as PropType<Array<GalleryItems>>
  }
})
</script>

<style lang="stylus" scoped>
@import '../styles/default.styl'

.gallery
  width 60%
  margin auto
  display flex
  flex-direction column
  flex-wrap wrap
  justify-content center
  align-items center

  .row
    width 100%
    background-color black
    justify-content center
    align-items center
    padding 0
    margin-bottom 50px
    border-radius 20px
    overflow hidden
    gap 0

  figure
    width 100%
    min-height 400px
    background-color blue
    text-align center

  header
    width 100%
    display flex
    flex-direction column
    justify-content center
    align-items center

    h2
      color green
</style>
