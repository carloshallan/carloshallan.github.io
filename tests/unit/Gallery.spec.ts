import { mount } from '@vue/test-utils'
import Gallery from '@/components/Gallery.vue'

describe('Gallery.vue', () => {
  it('Render gallery', () => {
    const gallery = mount(Gallery)
    expect(gallery.element).toMatchSnapshot()
  })
})
