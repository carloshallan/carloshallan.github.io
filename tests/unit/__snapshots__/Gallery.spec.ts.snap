// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery.vue Render gallery 1`] = `
<div
  class="gallery"
>
  <router-link
    class="row"
    tag="div"
    to="work"
  >
    <figure>
      Image
    </figure>
    <header>
      <h2>
        Title 1
      </h2>
      <p>
        test 1
      </p>
    </header>
  </router-link>
  <router-link
    class="row"
    tag="div"
    to="work"
  >
    <figure>
      Image
    </figure>
    <header>
      <h2>
        Title 2
      </h2>
      <p>
        test 2
      </p>
    </header>
  </router-link>
</div>
`;
